tests/integration/components/bs-form-test.js

```diff
@@ -159,6 +161,145 @@ module('Integration | Component | bs-form', function(hooks) {
+
+  test('Yielded #isSubmitting is never true if neither validate nor onSubmit returns Promise', async function(assert) {
+    this.set('submitAction', () => {});
+    this.set('validate', () => {});
+    this.set('hasValidator', true);
+    await this.render(hbs`{{#bs-form onSubmit=submitAction validate=validate hasValidator=hasValidator as |form|}}
+      <div class='state {{if form.isSubmitting 'is-submitting'}}'></div>
+    {{/bs-form}}`);
+
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+    triggerEvent('form', 'submit');
+  });
+
+  test('Yielded #isSubmitting is true as long as Promise returned by onSubmit is pending', async function(assert) {
+    this.set('submitAction', () => {
+      return new Promise((resolve) => {
+        later(() => resolve(), 10);
+      });
+    });
+    await this.render(hbs`{{#bs-form onSubmit=submitAction as |form|}}
+      <div class='state {{if form.isSubmitting 'is-submitting'}}'></div>
+    {{/bs-form}}`);
+
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+
+    triggerEvent('form', 'submit');
+    await waitFor('.is-submitting');
+    assert.dom('form .state').hasClass('is-submitting');
+
+    await settled();
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+  });
+
+  test('Yielded #isSubmitting is true as long as Promise returned by validate is pending', async function(assert) {
+    this.set('hasValidator', true);
+    this.set('validate', () => {
+      return new Promise((resolve) => {
+        later(() => resolve(), 10);
+      });
+    });
+    await this.render(hbs`{{#bs-form hasValidator=hasValidator validate=validate as |form|}}
+      <div class='state {{if form.isSubmitting 'is-submitting'}}'></div>
+    {{/bs-form}}`);
+
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+
+    triggerEvent('form', 'submit');
+    await waitFor('.is-submitting');
+    assert.dom('form .state').hasClass('is-submitting');
+
+    await settled();
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+  });
+
+  test('Yielded #isSubmitting is true as long as Promises returned by onSubmit and validate are pending', async function(assert) {
+    this.set('validate', () => {
+      return new Promise((resolve) => {
+        later(() => {
+          resolve();
+        }, 10);
+      });
+    });
+    this.set('submitAction', () => {
+      return new Promise((resolve) => {
+        later(() => {
+          resolve();
+        }, 10);
+      });
+    });
+    await this.render(hbs`{{#bs-form onSubmit=submitAction validate=validate hasValidator=true as |form|}}
+      <div class='state {{if form.isSubmitting 'is-submitting'}}'></div>
+    {{/bs-form}}`);
+
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+
+    triggerEvent('form', 'submit');
+    await waitFor('.is-submitting');
+    assert.dom('form .state').hasClass('is-submitting');
+
+    await new Promise((resolve) => {
+      later(() => resolve(), 11)
+    });
```

Comment:
Can you  please explain this? Shouldn't `submitAction` and `validate` be fulfilled after 11ms?